src/main/java/ch/njol/skript/expressions/ExprValueWithin.java

```diff
@@ -0,0 +1,98 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.classes.Changer;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.classes.ClassInfo;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.WrapperExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.Literal;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.registrations.Classes;
+import ch.njol.util.Kleenean;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+@Name("Value Within")
+@Description(
+	"Gets the value within objects. Usually used with variables to get the value they store rather than the variable itself, " +
+	"or with lists to get the values of a type."
+)
+@Examples({
+	"set {_entity} to a random entity out of all entities",
+	"delete entity within {_entity} # This deletes the entity itself and not the value stored in the variable",
+	"",
+	"set {_list::*} to \"something\", 10, \"test\" and a zombie",
+	"broadcast the strings within {_list::*} # \"something\", \"test\""
+})
+@Since("INSERT VERSION")
+public class ExprValueWithin extends WrapperExpression<Object> {
+
+	static {
+		Skript.registerExpression(ExprValueWithin.class, Object.class, ExpressionType.SIMPLE, "[the] (%-*classinfo%|value[s]) (within|in) %~objects%");
```

Comment:
WrapperExpressions in the past have been registered as ExpressionType.PROPERTY and only SIMPLE if it has no expression.

Honestly a new ExpressionType called WRAPPER should be made, but for now this should be PROPERTY.